### PR TITLE
[flutter_tools] remove mockito from mdns client test

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/mdns_discovery_test.dart [hunk 1]
```diff
@@ -254,17 +230,74 @@ void main() {
...
       );
-      final Uri uri = await portDiscovery.getObservatoryUri('bar', mockDevice, hostVmservicePort: 0);
+      final Uri uri = await portDiscovery.getObservatoryUri('bar', device, hostVmservicePort: 0);
       expect(uri.toString(), 'http://127.0.0.1:123/');
     });
   });
 }
 
-class MockMDnsClient extends Mock implements MDnsClient {}
+class FakeMDnsClient extends Fake implements MDnsClient {
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/mdns_discovery_test.dart:245]: When these `Fake`s get elaborate I'm always worried we're testing the fake class and not the actual interactions in prod.  If only we had a mocking framework...  But it's the same concern for complicated mocks.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/mdns_discovery_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/mdns_discovery_test.dart:245]: Its important to remember that unit tests really just exist to catch the most trivial of errors

Comment (jmagman) [reply packages/flutter_tools/test/general.shard/mdns_discovery_test.dart:245]: I'm just pontificating

**Pull request merged**